worlds/messenger/SubClasses.py

```diff
@@ -0,0 +1,58 @@
+from typing import Set, TYPE_CHECKING, Optional, Dict
+
+from BaseClasses import Region, Location, Item, ItemClassification, Entrance
+from .Constants import SEALS, NOTES, PROG_ITEMS, PHOBEKINS, USEFUL_ITEMS
+from .Options import Goal
+from .Regions import REGIONS
+
+if TYPE_CHECKING:
+    from . import MessengerWorld
+else:
+    MessengerWorld = object
+
+
+class MessengerRegion(Region):
+    def __init__(self, name: str, world: MessengerWorld):
+        super().__init__(name, world.player, world.multiworld)
+        self.add_locations(self.multiworld.worlds[self.player].location_name_to_id)
+        world.multiworld.regions.append(self)
+
+    def add_locations(self, name_to_id: Dict[str, int]) -> None:
+        for loc in REGIONS[self.name]:
+            self.locations.append(MessengerLocation(loc, self, name_to_id.get(loc, None)))
```

Comment:
because `REGIONS` is `Dict[str, Set]`, I believe this will generate different output if you roll the same seed twice.
Either use `sorted(REGIONS[self.name])` or change it to not be a set.

Reply:
ah for some reason i thought fill sorted these lmao oops

Reply:
fill does sort them, but since worlds can access the locations and items before the fill happens, it may still not be deterministic if you push your items in a non-deterministic order

Reply:
Hmm. My item creation was already deterministic and I don't do anything with my locations or items after creating them other than setting the rules, but the results actually were nondeterministic. I don't have any use for prefill atm just create everything and throw it at ap.